routers/api/v1/misc/signing.go

```diff
@@ -50,11 +52,70 @@ func SigningKey(ctx *context.APIContext) {
 		path = ctx.Repo.Repository.RepoPath()
 	}
 
-	content, err := asymkey_service.PublicSigningKey(ctx, path)
+	content, format, err := asymkey_service.PublicSigningKey(ctx, path)
 	if err != nil {
 		ctx.APIErrorInternal(err)
 		return
 	}
+	if format != git.KeyTypeOpenPGP {
+		ctx.APIErrorNotFound(errors.New("SSH keys are used for signing, not GPG"))
+		return
+	}
+	_, err = ctx.Write([]byte(content))
+	if err != nil {
+		ctx.APIErrorInternal(fmt.Errorf("Error writing key content %w", err))
+	}
```

Comment:
It should be something like this (old code is not right ...)

```suggestion
	if format != git.KeyTypeOpenPGP {
		ctx.APIErrorNotFound("SSH keys are used for signing, not GPG")
		return
	}
	_, _ = ctx.Write([]byte(content))
```

It shouldn't write any other response if the "Write" fails.